src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine_MapCreation.cs

```diff
@@ -145,48 +147,49 @@ private async Task<ProjectMap> CreateProjectMapAsync(ConcurrentSet<SymbolGroup>
+                            searchSymbol, _solution, projects, _options, cascadeDirection, cancellationToken).ConfigureAwait(false);
+                        AddSymbolTasks(result, symbols, symbolTasks, cancellationToken);
 
                         // Defer to the language to see if it wants to cascade here in some special way.
                         var symbolProject = _solution.GetProject(searchSymbol.ContainingAssembly);
                         if (symbolProject?.LanguageServices.GetService<ILanguageServiceReferenceFinder>() is { } service)
                         {
                             symbols = await service.DetermineCascadedSymbolsAsync(
-                                searchSymbol, symbolProject, cascadeDirection, _cancellationToken).ConfigureAwait(false);
-                            AddSymbolTasks(result, symbols, symbolTasks);
+                                searchSymbol, symbolProject, cascadeDirection, cancellationToken).ConfigureAwait(false);
+                            AddSymbolTasks(result, symbols, symbolTasks, cancellationToken);
                         }
 
-                        _cancellationToken.ThrowIfCancellationRequested();
+                        cancellationToken.ThrowIfCancellationRequested();
 
                         await Task.WhenAll(symbolTasks).ConfigureAwait(false);
-                    }, _cancellationToken, TaskCreationOptions.None, _scheduler).Unwrap());
+                    }, cancellationToken, TaskCreationOptions.None, _scheduler).Unwrap());
                 }
 
                 await Task.WhenAll(finderTasks).ConfigureAwait(false);
             }
         }
 
-        private async Task<SymbolGroup> DetermineSymbolGroupAsync(ISymbol searchSymbol)
+        private async Task<SymbolGroup> DetermineSymbolGroupAsync(ISymbol searchSymbol, CancellationToken cancellationToken)
         {
             if (!_options.Cascade)
                 return new SymbolGroup(ImmutableArray.Create(searchSymbol));
 
             return new SymbolGroup(
-                await SymbolFinder.FindLinkedSymbolsAsync(searchSymbol, _solution, _cancellationToken).ConfigureAwait(false));
+                await SymbolFinder.FindLinkedSymbolsAsync(searchSymbol, _solution, cancellationToken).ConfigureAwait(false));
         }
 
         private void AddSymbolTasks(
             ConcurrentSet<SymbolGroup> result,
             ImmutableArray<(ISymbol symbol, FindReferencesCascadeDirection cascadeDirection)> symbols,
-            ArrayBuilder<Task> symbolTasks)
+            ArrayBuilder<Task> symbolTasks,
+            CancellationToken cancellationToken)
         {
             if (!symbols.IsDefault)
             {
                 foreach (var (symbol, cascadeDirection) in symbols)
                 {
                     Contract.ThrowIfNull(symbol);
-                    _cancellationToken.ThrowIfCancellationRequested();
+                    cancellationToken.ThrowIfCancellationRequested();
```

Comment:
⚠️ In the event of cancellation,  the caller still needs to wait for tasks in `symbolTasks` to complete

Reply:
not sure i understand this comment.  can you clarify?

Reply:
At the point this exception is thrown, `symbolTasks` might contain one or more tasks which has not been awaited (not yet complete). The caller needs to wait for these tasks to complete as part of its cancellation handling.

Reply:
I see. Will address

Reply:
done.  this was a very good catch.  i'm wondering if this sort of mistake could be made easier to detect...